src/apps/yield-protocol/arbitrum/yield-protocol.pool.token-fetcher.ts

```diff
@@ -0,0 +1,102 @@
+import { Network } from '~types/network.interface';
+
+import { YieldProtocolContractFactory } from '../contracts';
+import { formatMaturity } from '../ethereum/yield-protocol.lend.token-fetcher';
+import { YIELD_PROTOCOL_DEFINITION } from '../yield-protocol.definition';
+
+const appId = YIELD_PROTOCOL_DEFINITION.id;
+const groupId = YIELD_PROTOCOL_DEFINITION.groups.pool.id;
+const network = Network.ARBITRUM_MAINNET;
+
+const YIELD_STRATEGIES = [
+  '0xE779cd75E6c574d83D3FD6C92F3CBE31DD32B1E1',
+  '0xE7214Af14BD70F6AAC9c16B0c1Ec9ee1CcC7EFDA',
+  '0x92A5B31310a3ED4546e0541197a32101fCfBD5c8',
+  '0xDC705FB403DBB93Da1d28388bc1DC84274593c11',
+];
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class ArbitrumYieldProtocolPoolTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(YieldProtocolContractFactory) private readonly yieldProtocolContractFactory: YieldProtocolContractFactory,
+  ) {}
+
+  async getPositions() {
+    const multicall = this.appToolkit.getMulticall(network);
+
+    const tokens = await Promise.all(
+      YIELD_STRATEGIES.map(async address => {
+        const strategyContract = this.yieldProtocolContractFactory.strategy({ address, network });
+
+        // strategy data
+        const [decimals, symbol, baseAddress, strategyTotalSupply, poolAddress] = await Promise.all([
+          multicall.wrap(strategyContract).decimals(),
+          multicall.wrap(strategyContract).symbol(),
+          multicall.wrap(strategyContract).base(),
+          multicall.wrap(strategyContract).totalSupply(),
+          multicall.wrap(strategyContract).pool(),
+        ]);
+
+        const poolContract = this.yieldProtocolContractFactory.pool({ address: poolAddress, network });
+
+        // pool data
+        const [baseReserves, fyTokenReserves, poolTotalSupply, maturity] = await Promise.all([
+          multicall.wrap(poolContract).getBaseBalance(),
+          multicall.wrap(poolContract).getFYTokenBalance(),
+          multicall.wrap(poolContract).totalSupply(),
+          multicall.wrap(poolContract).maturity(),
+        ]);
+
+        // get the corresponding base of the strategy
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+        const underlyingToken = baseTokens.find(v => v.address === baseAddress.toLowerCase());
+
+        if (!underlyingToken) return null;
+
+        // estimate the value of a unit of strategy token to base
+        const realFyTokenReserves = fyTokenReserves.sub(poolTotalSupply);
+        const estimate = (+baseReserves + +realFyTokenReserves) / +strategyTotalSupply;
+        const pricePerShare = estimate;
+        const price = pricePerShare * underlyingToken.price;
+
+        const displayProps: DisplayProps = {
+          label: `Yield ${underlyingToken.symbol} Strategy`,
+          secondaryLabel: `Automatic Roll on ${formatMaturity(maturity)}`,
```

Comment:
You might want to have `secondaryLabel` as the price of the token, and `tertiaryLabel` for your maturity.

Reply:
would like to keep as is because it is akin to a uni v2 pool (sort of), where the price isn't too meaningful for a user (they likely wouldn't act on that info).

Reply:
Got it 👍